src/Uno.UWP/Graphics/Display/DisplayInformation.cs

```diff
@@ -31,6 +35,43 @@ private DisplayInformation()
 			Initialize();
 		}
 
+		public static DisplayInformation GetForCurrentView()
+		{
+#if ANDROID
+			return GetForCurrentViewAndroid();
+#else
+			// This is needed to ensure for "current view" there is always a corresponding DisplayView instance.
+			// This means that Uno Islands and WinUI apps can keep using this API for now until we make the breaking change
+			// on Uno.WinUI codebase.
+			return GetOrCreateForWindowId(AppWindow.MainWindowId);
+#endif
+		}
+
+		internal static DisplayInformation GetForCurrentViewSafe() => GetForCurrentView();
+
+		internal static DisplayInformation GetForWindowId(WindowId windowId)
+		{
+			if (!_windowIdMap.TryGetValue(windowId, out var appView))
+			{
+				throw new InvalidOperationException(
+					$"ApplicationView corresponding with this window does not exist yet, which usually means " +
+					$"the API was called too early in the windowing lifecycle. Try to use ApplicationView later.");
+			}
+
+			return appView;
+		}
+
+		internal static DisplayInformation GetOrCreateForWindowId(WindowId windowId)
+		{
+			if (!_windowIdMap.TryGetValue(windowId, out var appView))
+			{
+				appView = new();
+				_windowIdMap[windowId] = appView;
+			}
+
+			return appView;
+		}
```

Comment:
It looks like all instances of `DisplayInformation` will just produce same results.